backend/src/ast/converters/scratch/__tests__/helpers.ts

```diff
@@ -0,0 +1,234 @@
+import { ActorNode, AstNodeType, GeneralAst } from "src/ast/types/general-ast";
+import {
+  CodeNode,
+  CodeNodeType,
+  EventListenerNode,
+} from "src/ast/types/general-ast/ast-nodes";
+import {
+  ExpressionNode,
+  ExpressionNodeType,
+} from "src/ast/types/general-ast/ast-nodes/code-node/expression-node";
+import ScratchInput from "src/ast/types/input/scratch";
+import { Block, Target } from "src/ast/types/input/scratch/generated/sb3";
+
+export const createScratchBlockInput = (
+  blocks: Target["blocks"],
+): ScratchInput => ({
+  targets: [
+    {
+      isStage: true,
+      name: "Stage",
+      variables: {},
+      lists: {},
+      broadcasts: {},
+      comments: {},
+      currentCostume: 0,
+      costumes: [
+        {
+          name: "backdrop1",
+          dataFormat: "svg",
+          assetId: "cd21514d0531fdffb22204e0ec5ed84a",
+          md5ext: "cd21514d0531fdffb22204e0ec5ed84a.svg",
+          rotationCenterX: 240,
+          rotationCenterY: 180,
+        },
+      ],
+      sounds: [],
+      volume: 100,
+      layerOrder: 0,
+      tempo: 60,
+      videoTransparency: 50,
+      videoState: "on",
+      textToSpeechLanguage: null,
+      blocks: blocks,
+    },
+  ],
+  monitors: [],
+  extensions: [],
+  meta: {
+    semver: "3.0.0",
+    vm: "4.5.471",
+    agent:
+      "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:131.0) Gecko/20100101 Firefox/131.0",
+  },
+});
+
+export const createScratchHatOutput = (
+  eventListenerNode: EventListenerNode,
+): GeneralAst => [
+  {
+    nodeType: AstNodeType.actor,
+    eventListeners: [eventListenerNode],
+    functionDeclarations: [],
+  } as ActorNode,
+];
+
+export const createScratchCodeInput = (
```

Comment:
From this function onwards, it's quite smart, but it lacks documentation.
Just a sentence saying `this allows us to generate a valid Scratch AST with arbitrary blocks, to support the AST conversion tests` would go a long way :-)